mailpoet/lib/WPCOM/DotcomLicenseProvisioner.php

```diff
@@ -0,0 +1,134 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\WPCOM;
+
+use MailPoet\API\JSON\ErrorResponse;
+use MailPoet\API\JSON\v1\Services;
+use MailPoet\API\JSON\v1\Settings;
+use MailPoet\Logging\LoggerFactory;
+use WP_Error;
+
+/**
+ * This class is responsible for receiving and activating the license purchased from WP.com Marketplace.
+ */
+class DotcomLicenseProvisioner {
+  const EVENT_TYPE_PROVISION_LICENSE = 'provision_license';
+
+  /** @var LoggerFactory */
+  private $loggerFactory;
+
+  /** @var Settings */
+  private $settings;
+
+  /** @var Services */
+  private $services;
+
+  public function __construct(
+    LoggerFactory $loggerFactory,
+    Settings $settings,
+    Services $services
+  ) {
+    $this->loggerFactory = $loggerFactory;
+    $this->settings = $settings;
+    $this->services = $services;
+  }
+
+  /**
+   * Returns true if in the context of WordPress.com Atomic platform.
+   *
+   * @return bool
+   */
+  public function isAtomicPlatform(): bool {
+    return defined('IS_ATOMIC') && IS_ATOMIC && defined('ATOMIC_CLIENT_ID') && (ATOMIC_CLIENT_ID === '2');
```

Comment:
I think that a little explanation at least why we need to check `ATOMIC_CLIENT_ID === '2'` would be helpful. If I remember correctly ATOMIC_CLIENT_ID 2 is client for all customer sites and there are other clients for internal sites. Am I correct?